### ISSUE TITLE
Incorrect parsing when using `span` as child element in EPUB 3 navigation document

### ISSUE THREAD

**Issue by kianmeng** [johnfactotum/foliate#781]
**Describe the bug**

Foliate cannot show the nested submenu items in the table of content menu list.

**To Reproduce**
1. Rename `no_nested_submenu_in_toc.zip ` to `no_nested_submenu_in_toc.epub` 
2. Open `no_nested_submenu_in_toc.epub` with Foliate.
3. Click the table of content menu button.
4. See screenshot.

**Expected behavior**

**Screenshots**
In Foliate:
![image](https://user-images.githubusercontent.com/134518/132935376-dc6da00d-59e9-45f0-a0c8-e7795e905dac.png)

In Calibre Viewer:
![image](https://user-images.githubusercontent.com/134518/132935328-3d84fa9b-1607-4647-a055-e85cf35efc13.png)

**Version:**
 - Foliate version: 2.6.3
 - OS/Distribution and version: Ubuntu 21.04
 - Desktop environment: Gnome 3.38.4
 - Installation method: Flatpak

**Additional context**

Sample epub file as attached. Please rename to `.epub` to test.
[no_nested_submenu_in_toc.zip](https://github.com/johnfactotum/foliate/files/7147159/no_nested_submenu_in_toc.zip)

Stacktrace:

```bash
$ gjs /usr/bin/com.github.johnfactotum.Foliate no_nested_submenu_in_toc.epub 

(WebKitWebProcess:2): dbind-WARNING **: 11:48:29.114: Couldn't connect to accessibility bus: Failed to connect to socket /tmp/dbus-tVNGTPFeb6: No such file or directory

(com.github.johnfactotum.Foliate:198490): Gjs-WARNING **: 11:48:35.987: Unhandled promise rejection. To suppress this warning, add an error handler to your promise chain with .catch() or a try-catch block around your await expression. Stack trace of the failed promise:
  main@resource:///com/github/johnfactotum/Foliate/js/main.js:476:24
  run@resource:///org/gnome/gjs/modules/script/package.js:206:19
  @/usr/bin/com.github.johnfactotum.Foliate:9:17
```

**Comment by johnfactotum** [comment]
Will need to investigate further, but I think this might be a regression with the latest changes of Epub.js, possibly related to https://github.com/futurepress/epub.js/pull/1114.

It probably only happens for undefined or duplicate id or href, since I haven't noticed any issues for nested toc items with most books. Even older versions of Epub.js doesn't parse this toc correctly, though (it drops the "Languages" item and promotes its subitem to the top level).

**Comment by johnfactotum** [comment]
OK, I think this is because Epub.js doesn't support using `span` in the nav item, and instead requires `a`.

https://github.com/futurepress/epub.js/blob/228d9913e045a0515f5d22637178384850d8810a/src/navigation.js#L194-L198

The difference is that after futurepress/epub.js#1114, it traverses the TOC recursively, instead of visiting each `li` sequentially. So it drops the whole branch at "Languages" because it doesn't contain an `a`. Previously, since it processes all `li`s individually, it doesn't drop those subitems, but it still ignores "Languages" so it was already broken.

It appears, though, that Epub.js v2 actually does handle `span`s. So it is a regression, one that happened way back in some version of v3.

**Comment by kianmeng** [comment]
@johnfactotum If I understand correctly, we'll have to wait for the upstream `epub.js` to fix it then?

**Comment by johnfactotum** [comment]
Yes. I will try to make a pull request if/when I have time.

**Comment by kianmeng** [comment]
> Yes. I will try to make a pull request if/when I have time.

@johnfactotum Appreciate it and thank you very much! :+1:

**Comment by johnfactotum** [comment]
Forgot to close this. This has been fixed with e867c214b12a6dea57d626d2d00a19312a43a837.

**Comment by kianmeng** [comment]
:tada: :tada: :tada: :tada: :tada:
